Ok now we see here the same error that @pablobecker was obtaining when trying to merge master. 

```
[ 16%] Building CXX object kratos/CMakeFiles/KratosCore.dir/Unity/unity_8_cxx.cxx.o
In file included from /__w/Kratos/Kratos/kratos/tests/cpp_tests/sources/test_filesystem.cpp:21,
                 from /__w/Kratos/Kratos/build/Custom/kratos/CMakeFiles/KratosCore.dir/Unity/unity_8_cxx.cxx:28:
/__w/Kratos/Kratos/kratos/includes/kratos_filesystem.h:43:76: error: 'deprecated' attribute directive ignored [-Werror=attributes]
 namespace KRATOS_DEPRECATED_MESSAGE("Please use std::filesystem directly") filesystem {
                                                                            ^~~~~~~~~~
cc1plus: all warnings being treated as errors
```

It seems that this error is only fixed in gcc 10 https://gcc.gnu.org/bugzilla/show_bug.cgi?id=79817

Should we skip this warning?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -193,7 +193,7 @@ jobs:
 
     - name: Build
       run: |
-        export KRATOS_CMAKE_CXX_FLAGS="-Werror -Wno-maybe-uninitialized -Wno-deprecated-declarations -Wignored-qualifiers"
+        export KRATOS_CMAKE_CXX_FLAGS="-Werror -Wno-deprecated-declarations -Wignored-qualifiers"
         cp .github/workflows/centos_configure.sh centos_configure.sh
         bash centos_configure.sh
 

--- a/kratos/includes/kratos_filesystem.h
+++ b/kratos/includes/kratos_filesystem.h
@@ -34,15 +34,15 @@
 
 namespace Kratos {
 
-// deprecated namespaces wrongly issue a warning with GCC 9, hence disabling until removed
-#if defined(__GNUG__) && __GNUC__ == 9 && !defined(__clang__) && !defined(__INTEL_COMPILER)
+// deprecated namespaces wrongly issue a warning with GCC < 10, hence disabling until removed
+#if defined(__GNUG__) && __GNUC__ < 10 && !defined(__clang__) && !defined(__INTEL_COMPILER)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wattributes"
 #endif
 
 namespace KRATOS_DEPRECATED_MESSAGE("Please use std::filesystem directly") filesystem {
 
-#if defined(__GNUG__) && __GNUC__ == 9 && !defined(__clang__) && !defined(__INTEL_COMPILER)
+#if defined(__GNUG__) && __GNUC__ < 10 && !defined(__clang__) && !defined(__INTEL_COMPILER)
 #pragma GCC diagnostic pop
 #endif
 

